21:57:10:| Object of type SimpleNamespace is not JSON serializable
Message:/newhirn
Except:Traceback (most recent call last):
  File "C:\Users\lachl\Repos\GitHub\telebagger\telebagger.py", line 160, in my_event_handler
    await self.telegram_command(signal)
  File "C:\Users\lachl\Repos\GitHub\telebagger\telebagger.py", line 110, in telegram_command
    await new_signal.new_signal(signal, self.trade_stream)
  File "C:\Users\lachl\Repos\GitHub\telebagger\new_signal.py", line 35, in new_signal
    trades = await get_trades(signal)
  File "C:\Users\lachl\Repos\GitHub\telebagger\new_signal.py", line 23, in get_trades
    response = requests.post(url, json=data)
  File "C:\Users\lachl\Repos\GitHub\telebagger\.venv\lib\site-packages\requests\api.py", line 119, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "C:\Users\lachl\Repos\GitHub\telebagger\.venv\lib\site-packages\requests\api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "C:\Users\lachl\Repos\GitHub\telebagger\.venv\lib\site-packages\requests\sessions.py", line 528, in request
    prep = self.prepare_request(req)
  File "C:\Users\lachl\Repos\GitHub\telebagger\.venv\lib\site-packages\requests\sessions.py", line 456, in prepare_request
    p.prepare(
  File "C:\Users\lachl\Repos\GitHub\telebagger\.venv\lib\site-packages\requests\models.py", line 319, in prepare
    self.prepare_body(data, files, json)
  File "C:\Users\lachl\Repos\GitHub\telebagger\.venv\lib\site-packages\requests\models.py", line 469, in prepare_body
    body = complexjson.dumps(json)
  File "C:\Python310\lib\json\__init__.py", line 231, in dumps
    return _default_encoder.encode(obj)
  File "C:\Python310\lib\json\encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "C:\Python310\lib\json\encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "C:\Python310\lib\json\encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type SimpleNamespace is not JSON serializable


22:13:36:| 'SpotBasic' object has no attribute 'id'
Message:/newhirn
Except:Traceback (most recent call last):
  File "C:\Users\lachl\Repos\GitHub\telebagger\telebagger.py", line 160, in my_event_handler
    await self.telegram_command(signal)
  File "C:\Users\lachl\Repos\GitHub\telebagger\telebagger.py", line 110, in telegram_command
    await new_signal.new_signal(signal, self.trade_stream)
  File "C:\Users\lachl\Repos\GitHub\telebagger\new_signal.py", line 36, in new_signal
    trades = await get_trades(signal)
  File "C:\Users\lachl\Repos\GitHub\telebagger\new_signal.py", line 21, in get_trades
    print('posting hirn signal', signal[0])
  File "C:\Users\lachl\Repos\GitHub\telebagger\trade_conditions.py", line 57, in __str__
    'id': self.id,
AttributeError: 'SpotBasic' object has no attribute 'id'


22:14:41:| __str__ returned non-string (type dict)
Message:/newhirn
Except:Traceback (most recent call last):
  File "C:\Users\lachl\Repos\GitHub\telebagger\telebagger.py", line 160, in my_event_handler
    await self.telegram_command(signal)
  File "C:\Users\lachl\Repos\GitHub\telebagger\telebagger.py", line 110, in telegram_command
    await new_signal.new_signal(signal, self.trade_stream)
  File "C:\Users\lachl\Repos\GitHub\telebagger\new_signal.py", line 36, in new_signal
    trades = await get_trades(signal)
  File "C:\Users\lachl\Repos\GitHub\telebagger\new_signal.py", line 21, in get_trades
    print('posting hirn signal', signal[0])
TypeError: __str__ returned non-string (type dict)


22:23:08:| dictionary update sequence element #0 has length 1; 2 is required
Message:/hirn
Except:Traceback (most recent call last):
  File "C:\Users\lachl\Repos\GitHub\telebagger\telebagger.py", line 160, in my_event_handler
    await self.telegram_command(signal)
  File "C:\Users\lachl\Repos\GitHub\telebagger\telebagger.py", line 97, in telegram_command
    await new_signal.new_signal(signal, self.trade_stream)
  File "C:\Users\lachl\Repos\GitHub\telebagger\new_signal.py", line 36, in new_signal
    trades = await get_trades(signal)
  File "C:\Users\lachl\Repos\GitHub\telebagger\new_signal.py", line 22, in get_trades
    data = json.dumps(dict(str(signal[0])))
ValueError: dictionary update sequence element #0 has length 1; 2 is required


22:23:58:| dictionary update sequence element #0 has length 1; 2 is required
Message:/newhirn
Except:Traceback (most recent call last):
  File "C:\Users\lachl\Repos\GitHub\telebagger\telebagger.py", line 160, in my_event_handler
    await self.telegram_command(signal)
  File "C:\Users\lachl\Repos\GitHub\telebagger\telebagger.py", line 110, in telegram_command
    await new_signal.new_signal(signal, self.trade_stream)
  File "C:\Users\lachl\Repos\GitHub\telebagger\new_signal.py", line 36, in new_signal
    trades = await get_trades(signal)
  File "C:\Users\lachl\Repos\GitHub\telebagger\new_signal.py", line 22, in get_trades
    data = dict(str(signal[0]))
ValueError: dictionary update sequence element #0 has length 1; 2 is required


22:25:09:| Expecting property name enclosed in double quotes: line 1 column 2 (char 1)
Message:/newhirn
Except:Traceback (most recent call last):
  File "C:\Users\lachl\Repos\GitHub\telebagger\telebagger.py", line 160, in my_event_handler
    await self.telegram_command(signal)
  File "C:\Users\lachl\Repos\GitHub\telebagger\telebagger.py", line 110, in telegram_command
    await new_signal.new_signal(signal, self.trade_stream)
  File "C:\Users\lachl\Repos\GitHub\telebagger\new_signal.py", line 36, in new_signal
    trades = await get_trades(signal)
  File "C:\Users\lachl\Repos\GitHub\telebagger\new_signal.py", line 22, in get_trades
    data = json.loads((str(signal[0])))
  File "C:\Python310\lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "C:\Python310\lib\json\decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "C:\Python310\lib\json\decoder.py", line 353, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)


22:26:11:| Expecting property name enclosed in double quotes: line 1 column 2 (char 1)
Message:/newhirn
Except:Traceback (most recent call last):
  File "C:\Users\lachl\Repos\GitHub\telebagger\telebagger.py", line 160, in my_event_handler
    await self.telegram_command(signal)
  File "C:\Users\lachl\Repos\GitHub\telebagger\telebagger.py", line 110, in telegram_command
    await new_signal.new_signal(signal, self.trade_stream)
  File "C:\Users\lachl\Repos\GitHub\telebagger\new_signal.py", line 37, in new_signal
    trades = await get_trades(signal)
  File "C:\Users\lachl\Repos\GitHub\telebagger\new_signal.py", line 23, in get_trades
    data = json.loads((str(signal[0])))
  File "C:\Python310\lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "C:\Python310\lib\json\decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "C:\Python310\lib\json\decoder.py", line 353, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)


22:28:09:| Expecting property name enclosed in double quotes: line 1 column 2 (char 1)
Message:/newhirn
Except:Traceback (most recent call last):
  File "C:\Users\lachl\Repos\GitHub\telebagger\telebagger.py", line 160, in my_event_handler
    await self.telegram_command(signal)
  File "C:\Users\lachl\Repos\GitHub\telebagger\telebagger.py", line 110, in telegram_command
    await new_signal.new_signal(signal, self.trade_stream)
  File "C:\Users\lachl\Repos\GitHub\telebagger\new_signal.py", line 37, in new_signal
    trades = await get_trades(signal)
  File "C:\Users\lachl\Repos\GitHub\telebagger\new_signal.py", line 23, in get_trades
    data = json.loads((str(signal[0])))
  File "C:\Python310\lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "C:\Python310\lib\json\decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "C:\Python310\lib\json\decoder.py", line 353, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)


22:29:31:| Expecting property name enclosed in double quotes: line 1 column 2 (char 1)
Message:/newhirn
Except:Traceback (most recent call last):
  File "C:\Users\lachl\Repos\GitHub\telebagger\telebagger.py", line 160, in my_event_handler
    await self.telegram_command(signal)
  File "C:\Users\lachl\Repos\GitHub\telebagger\telebagger.py", line 110, in telegram_command
    await new_signal.new_signal(signal, self.trade_stream)
  File "C:\Users\lachl\Repos\GitHub\telebagger\new_signal.py", line 37, in new_signal
    trades = await get_trades(signal)
  File "C:\Users\lachl\Repos\GitHub\telebagger\new_signal.py", line 23, in get_trades
    data = json.loads((str(signal[0])))
  File "C:\Python310\lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "C:\Python310\lib\json\decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "C:\Python310\lib\json\decoder.py", line 353, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)


22:35:24:| 'SpotBasic' object has no attribute 'get_dict'
Message:/newhirn
Except:Traceback (most recent call last):
  File "C:\Users\lachl\Repos\GitHub\telebagger\telebagger.py", line 160, in my_event_handler
    await self.telegram_command(signal)
  File "C:\Users\lachl\Repos\GitHub\telebagger\telebagger.py", line 110, in telegram_command
    await new_signal.new_signal(signal, self.trade_stream)
  File "C:\Users\lachl\Repos\GitHub\telebagger\new_signal.py", line 37, in new_signal
    trades = await get_trades(signal)
  File "C:\Users\lachl\Repos\GitHub\telebagger\new_signal.py", line 23, in get_trades
    data = signal[0].get_dict()
AttributeError: 'SpotBasic' object has no attribute 'get_dict'


